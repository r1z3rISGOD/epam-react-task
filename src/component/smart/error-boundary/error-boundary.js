"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ErrorBoundary = void 0;
var React = require("react");
var ErrorBoundary = function (_a) {
    var children = _a.children;
    var OopsText = function () { return (React.createElement("h2", null, "Something went wrong!")); };
    var isEverythingOk = true;
    return React.createElement(React.Fragment, null, isEverythingOk ? children : React.createElement(OopsText, null));
};
exports.ErrorBoundary = ErrorBoundary;
//# sourceMappingURL=error-boundary.js.map