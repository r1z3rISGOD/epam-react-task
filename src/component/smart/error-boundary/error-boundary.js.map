{"version":3,"file":"error-boundary.js","sourceRoot":"","sources":["error-boundary.tsx"],"names":[],"mappings":";;;AAAA,6BAA8B;AAOvB,IAAM,aAAa,GAAkC,UAAC,EAAgB;QAAd,QAAQ,cAAA,EAAE,EAAE,QAAA;IACzE,IAAM,QAAQ,GAAG,cAAM,OAAA,CACjB,wDAEK,CACV,EAJsB,CAItB,CAAA;IAED,OAAO,0CAAG,EAAE,CAAC,CAAC,CAAC,QAAQ,CAAC,CAAC,CAAC,oBAAC,QAAQ,OAAG,CAAI,CAAA;AAC5C,CAAC,CAAA;AARY,QAAA,aAAa,iBAQzB"}