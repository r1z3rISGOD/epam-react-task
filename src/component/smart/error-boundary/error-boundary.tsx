import * as React from 'react'

interface ErrorBoundaryProps {
    children: React.ReactNode
    ok: boolean
}

export const ErrorBoundary : React.FC<ErrorBoundaryProps> = ({ children, ok }) => {
  const OopsText = () => (
        <h2>
            Something went wrong!
        </h2>
  )

  return <>{ok ? children : <OopsText />}</>
}
