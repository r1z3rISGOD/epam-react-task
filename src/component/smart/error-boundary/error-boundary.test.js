import { ErrorBoundary } from './error-boundary.tsx'
import React from 'react'

const child = <div>Hello</div>

describe('error boundary tests', () => {
  it('error boundary test 1', () => {
    const comp = shallow(<ErrorBoundary children={child} ok={true}/>)
    expect(comp).toMatchSnapshot()
  })
  it('error boundary test 2', () => {
    const comp = shallow(<ErrorBoundary children={child} ok={false}/>)
    expect(comp).toMatchSnapshot()
  })
})
