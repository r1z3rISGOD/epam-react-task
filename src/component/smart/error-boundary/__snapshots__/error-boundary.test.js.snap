// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`error boundary tests error boundary test 1 1`] = `
<Fragment>
  <div>
    Hello
  </div>
</Fragment>
`;

exports[`error boundary tests error boundary test 2 1`] = `
<Fragment>
  <OopsText />
</Fragment>
`;
